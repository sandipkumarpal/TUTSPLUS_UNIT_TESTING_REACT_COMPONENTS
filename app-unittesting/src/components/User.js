import React from "react";

function User(props) {
    return (
        <li className="list-group-item">{props.user.name}</li>
    );
}

export default User;
