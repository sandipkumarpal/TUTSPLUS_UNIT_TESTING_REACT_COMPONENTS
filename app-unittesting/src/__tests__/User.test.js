import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow} from 'enzyme';
import User from '../components/User';

const userData = {
    name: 'Scott'
};

test('<User /> render correctly', () => {
    const tree = renderer
        .create(<User key={userData.name} user={userData}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
