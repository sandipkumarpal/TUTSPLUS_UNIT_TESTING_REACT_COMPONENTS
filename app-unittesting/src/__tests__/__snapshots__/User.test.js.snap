// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> render correctly 1`] = `
<li
  className="list-group-item"
>
  Scott
</li>
`;
